CNV-17271: RN Adding known issues for 4.11 (#48328)
openshift/openshift-docs: virt/virt-4-11-release-notes.adoc

[id="virt-4-11-known-issues"]
== Known issues

* Some persistent volume claim (PVC) annotations created by the Containerized Data Importer (CDI) can cause the virtual machine snapshot restore operation to hang indefinitely. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2070366[*BZ#2070366*])
** As a workaround, you can remove the annotations manually:
. Obtain the xref:../virt/virtual_machines/virtual_disks/virt-managing-vm-snapshots.adoc#virtual-machine-snapshot-controller-and-custom-resource-definitions-crds[VirtualMachineSnapshotContent] custom resource (CR) name from the `status.virtualMachineSnapshotContentName` value in the `VirtualMachineSnapshot` CR.

> . Edit the `VirtualMachineSnapshotContent` CR by removing all lines that contain `k8s.io/cloneRequest`.

Reviewer: Suggestion for clarity (unsure if this is just a Me Thing):

```
. Edit the `VirtualMachineSnapshotContent` CR and remove all lines that contain `k8s.io/cloneRequest`.
```

(If this were a normal procedure, I'd also want to see steps for opening/viewing these CRs, but because it's a known issue, I think this is ok.)

> . Optional: Delete the `DataVolume` and `PersistentVolumeClaim` objects in this namespace if their name begins with `restore-` and they are not referenced by virtual machines. This step is required if you did not specify a value for `spec.dataVolumeTemplates` in the `VirtualMachine` object.

Reviewer: Nitpicky suggestion (take or leave it) because I'm not sure if people will read the whole step if it starts with "Optional":

```suggestion
. If you did not specify a value for `spec.dataVolumeTemplates` in the `VirtualMachine` object, delete any `DataVolume` and `PersistentVolumeClaim` objects in this namespace where both of the following conditions are true:
* The object's name begins with `restore-`.
* The object is not referenced by virtual machines. 
+
This step is optional if you specified a value for `spec.dataVolumeTemplates`.
```

. Repeat the xref:../virt/virtual_machines/virtual_disks/virt-managing-vm-snapshots.adoc#virt-restoring-vm-from-snapshot-cli_virt-managing-vm-snapshots[restore operation] with the updated `VirtualMachineSnapshot` CR.

> * The restore process of a cloned virtual machine (VM) snapshot fails if the source VM is deleted. This is because the `DataVolumeTemplate` value in the `VirtualMachineSnapshotContent` CR which refers to the persistent volume claim (PVC) of the source VM triggers a creation of clone data volume from a non existing source and is stuck waiting on the source PVC to be ready. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2104479[*BZ#2104479*])

Reviewer: I'm having trouble following this second sentence, and I'm not sure that all of the technical details are necessary to get the point across. Maybe:

```suggestion
* The restore process of a cloned virtual machine (VM) snapshot fails if the source VM is deleted. This is because the `DataVolumeTemplate` value in the `VirtualMachineSnapshotContent` CR refers to the source VM's PVC, which no longer exists. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2104479[*BZ#2104479*])
```
